x-pack/plugins/canvas/public/components/workpad_loader/index.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import moment from 'moment';
+// @ts-expect-error
+import { getDefaultWorkpad } from '../../state/defaults';
+import { canUserWrite } from '../../state/selectors/app';
+import { getWorkpad } from '../../state/selectors/workpad';
+import { getId } from '../../lib/get_id';
+import { downloadWorkpad } from '../../lib/download_workpad';
+import { ComponentStrings, ErrorStrings } from '../../../i18n';
+import { State, CanvasWorkpad } from '../../../types';
+import { useServices } from '../../services';
+// @ts-expect-error
+import { WorkpadLoader as Component } from './workpad_loader';
+
+const { WorkpadLoader: strings } = ComponentStrings;
+const { WorkpadLoader: errors } = ErrorStrings;
+
+type WorkpadStatePromise = ReturnType<ReturnType<typeof useServices>['workpad']['find']>;
+type WorkpadState = WorkpadStatePromise extends PromiseLike<infer U> ? U : never;
+
+export const WorkpadLoader: FC<{ onClose: () => void }> = ({ onClose }) => {
+  const fromState = useSelector((state: State) => ({
+    workpadId: getWorkpad(state).id,
+    canUserWrite: canUserWrite(state),
+  }));
+
+  const [workpadsState, setWorkpadsState] = useState<WorkpadState | null>(null);
+  const services = useServices();
```

Comment:
Switch to specific hooks, makes the code easier to read and the `useEffect` dependencies below easier.